nixos/modules/programs/foot/default.nix

```diff
@@ -0,0 +1,92 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.foot;
+
+  settingsFormat = pkgs.formats.ini {
+    listsAsDuplicateKeys = true;
+    mkKeyValue =
+      with lib.generators;
+      mkKeyValueDefault {
+        mkValueString =
+          v:
+          mkValueStringDefault { } (
+            if v == true then
+              "yes"
+            else if v == false then
+              "no"
+            else if v == null then
+              "none"
+            else
+              v
+          );
+      } "=";
+  };
+in
+{
+  options.programs.foot = {
+    enable = lib.mkEnableOption "foot terminal emulator";
+
+    package = lib.mkPackageOption pkgs "foot" { };
+
+    settings = lib.mkOption {
+      inherit (settingsFormat) type;
+      default = { };
+      description = ''
+        Configuration for foot terminal emulator. Further information can be found in {command}`man 5 foot.ini`.
+
+        Global configuration has to be written under the [main] section.
+      '';
+      example = {
+        main.font = "FreeMono:size=12";
+        scrollback.lines = 100000;
+      };
+    };
+
+    theme = lib.mkOption {
+      type = with lib.types; nullOr str;
+      default = null;
+      description = ''
+        Theme name. Check <https://codeberg.org/dnkl/foot/src/branch/master/themes> for available themes.
+      '';
+      example = "aeroroot";
+    };
+
+    enableBashIntegration = lib.mkEnableOption "foot bash integration" // {
+      default = true;
+    };
+
+    enableFishIntegration = lib.mkEnableOption "foot fish integration" // {
+      default = true;
+    };
+
+    enableZshIntegration = lib.mkEnableOption "foot zsh integration" // {
+      default = true;
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment = {
+      systemPackages = [ cfg.package ];
+      etc."xdg/foot/foot.ini".source = settingsFormat.generate "foot.ini" cfg.settings;
+    };
+    programs = {
+      foot.settings.main.include = lib.optionals (cfg.theme != null) [
+        "${pkgs.foot.themes}/share/foot/themes/${cfg.theme}"
+      ];
+      # https://codeberg.org/dnkl/foot/wiki#user-content-shell-integration
```

Comment:
@sternenseemann 

> Where do the shell integration snippets come from

Please see the link above :)

> and what license do they have?

Nice catch.  I am not sure about that.

> I'm also not sure if it is the best behavior automatically injecting extra code into user's shells without them requesting it…

We had an discussion here: https://github.com/NixOS/nixpkgs/pull/333390#discussion_r1734610257.  I would say it is the convention.  As I said, I do not have a strong opinion on this.  Free feel to revert it if you want.

Reply:
https://codeberg.org/dnkl/foot/issues/1821